### PR TITLE
Kintsugi: Support Git merge driver

### PR DESCRIPTION
**Pull request by byohay**
On top of #17 

- kintusgi.rb: Support three way merge.
- kintsugi: Support merge driver.
- README.md: Explain how to use Kintsugi as merge driver.

### CODE REVIEW

File: bin/kintsugi [hunk 1]
```diff
@@ -10,43 +10,101 @@ require "optparse"
...
+      exit(1)
     end
+    Kintsugi.three_way_merge(arguments[0], arguments[1], arguments[2], arguments[3])
+  }
+
+COMMANDS = {
+  "driver" => Command.new(
+    option_parser: DRIVER_OPTION_PARSER,
+    action: DRIVER_ACTION,
+    description: "Git merge driver mode"
```
Comment (ashdnazg) [inline bin/kintsugi:42]: IMO, a subcommand is not a mode, I would go with a verb `3-way merge compatible with Git merge driver`
or similar.

File: README.md [hunk 2]
```diff
@@ -38,6 +38,25 @@ When there's a `.pbxproj` file with Git conflicts, and a 3-way merge is possible
...
+```
+
+`kintsugi` should be in your `PATH`.
+Run `git config` with `--global` to add this to the global config file.
+
+- Set the the merge driver of the repository to use Kintsugi for `.pbxproj` files, by adding the following line to the `.gitattributes` file of the repository:
+
+`*.pbxproj merge=kintsugi`
+
+You can set this globally by putting it in `$HOME/.config/git/attributes`.
```
Comment (ashdnazg) [inline README.md:58]: be more accurate about the path of git attributes (you can still keep an example) and link to the official git doc page.

File: bin/kintsugi [hunk 3]
```diff
@@ -10,43 +10,101 @@ require "optparse"
...
+    puts option_parser
+    exit(1)
+  end
+
+  project_file_path = File.expand_path(arguments[0])
+  Kintsugi.resolve_conflicts(project_file_path, option[:"changes-output-path"])
+  puts "Resolved conflicts successfully"
+}
+
+ROOT_COMMAND = Command.new(
```
Comment (ashdnazg) [inline bin/kintsugi:84]: Extract all constants and maybe other things as well to `lib`

**Review by ashdnazg — CHANGES_REQUESTED** [review]

**Comment by byohay** [comment]
@ashdnazg round done.

File: lib/kintsugi/cli.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+# frozen_string_literal: true
+
+require "optparse"
+
+require_relative "version"
+
+module Kintsugi
+  module CLI
+    Command = Struct.new(:option_parser, :action, :description, keyword_init: true)
+
```
Comment (ashdnazg) [inline lib/kintsugi/cli.rb:12]: not sure if you can make stuff private, but in any case put some documentation in this file as currently it has no context.

File: README.md [hunk 5]
```diff
@@ -44,18 +44,20 @@ To use Kintsugi as a Git merge driver, follow these steps:
 
 - Add it as driver to Git config file by running the following:
 ```sh
-git config merge.kintsugi.name "Kintsugi driver" # Or any other name that your prefer
+git config merge.kintsugi.name "Kintsugi driver" # Or any other name you prefer
```
Comment (ashdnazg) [inline README.md:47]: reconsider whether the probable use is setting this for a specific repository or making it global.

**Review by ashdnazg — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (byohay) [reply README.md:47]: I still prefer to have the local case the default :)

**Comment by byohay** [comment]
@ashdnazg squashed.

**Review by ashdnazg — APPROVED** [review]
Test that it works and merge

**Pull request merged**